source/AI.cpp

```diff
@@ -1719,19 +1772,21 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, const System *playerSy
 			parent.Attributes().Get("fuel capacity") - maxFuel)
 		return true;
 
-	// If an out-of-combat NPC carried ship is carrying a significant cargo
+	// If an out-of-combat carried ship is carrying a significant cargo
 	// load and can transfer some of it to the parent, it should do so.
-	if(!ship.IsYours())
-	{
-		bool hasEnemy = ship.GetTargetShip() && ship.GetTargetShip()->GetGovernment()->IsEnemy(ship.GetGovernment());
-		if(!hasEnemy)
-		{
-			const CargoHold &cargo = ship.Cargo();
-			// Mining ships only mine while they have 5 or more free space. While mining, carried ships
-			// do not consider docking unless their parent is far from a targetable asteroid.
-			if(parent.Cargo().Free() && !cargo.IsEmpty() && cargo.Size() && cargo.Free() < 5)
-				return true;
-		}
+	bool hasEnemy = ship.GetTargetShip() && ship.GetTargetShip()->GetGovernment()->IsEnemy(ship.GetGovernment());
+	if(!hasEnemy)
+	{
+		const CargoHold &cargo = ship.Cargo();
+		// NPC ships should always transfer cargo.
+		bool shouldTransferCargo = true;
+		// Player ships should only transfer cargo if they set the AI preference.
+		if(ship.IsYours())
+			shouldTransferCargo = Preferences::Has("Fighters transfer cargo");
+		// Mining ships only mine while they have 5 or more free space. While mining, carried ships
+		// do not consider docking unless their parent is far from a targetable asteroid.
+		if(shouldTransferCargo && parent.Cargo().Free() && !cargo.IsEmpty() && cargo.Size() && cargo.Free() < 5)
+			return true;
```

Comment:
Instead of checking the preference last, why not check the preference first (just like it is done with the original code but with `!ship.IsYours()`)?

Reply:
What do you mean it is checked last?  It is first in the ordering or do you mean right to left is first in ordering?

Reply:
Sorry, I was referring to the hasEnemy checks.

Reply:
I think I understand.  I can experiment and push what I find.

Reply:
I pushed 128211223faf80e030cd0c11f0bae10f168a3baa and after some discord conversation with @quyykk I also pushed ac939ccf86f53e065eae5a4f580e3bf1b16766e4.  This feedback is resolved.